src/faq/troubleshooting.rst

```diff
@@ -0,0 +1,222 @@
+very worst, you'll have multiple users attempting to claim a single ticket, and
+if you've grabbed several of those from your view, you simply move on to the
+next ticket and try again.
+
+.. note::
+    This FAQ entry was borrowed from
+    http://stackoverflow.com/questions/299723/can-i-do-transactions-and-locks-in-couchdb
+    with permission from the author
+
+Why does creating my view take so long?
+---------------------------------------
+
+There are a number of possible reasons:
+
+1. Your reduce function is not reducing the input data to a small enough output.
+See Introduction_to_CouchDB_views#reduce_functions for more details.
+
+2. If you have a lot of documents or lots of large documents (going into the
+millions and Gigabytes), the first time a view index is created just takes the
+time it is needed to run through all documents.
+
+3. If you use the emit()-function in your view with doc as the second parameter
+you effectively copy your entire data into the view index. This takes a lot of
+time. Consider rewriting your emit() call to emit(key, null); and query the view
+with the `?include_docs=true parameter` to get all doc's data with the view
+without having to copy the data into the view index.
+
+How can I "JOIN" with CouchDB?
+------------------------------
+
+See https://www.cmlenz.net/archives/2007/10/couchdb-joins
+
+Why is my database so large, even after compaction?
+---------------------------------------------------
+
+Often, CouchDB users expect that adding a document to a database, then deleting
+that document will return the database to its original state. However, this is
+not the case. Consider a two-database scenario:
+
+- Doc 1 inserted to DB A.
+- DB A replicated to DB B.
+- Doc 1 deleted from DB A.
+- DB A replicated to DB B.
+
+If inserting and then deleting a document returned the database to the original
+state, the second replication from A to B would be "empty" and hence DB B would
+be unchanged, which means it would be out of sync with DB A.
+
+To handle this case, CouchDB keeps a record of each document deleted, by keeping
+the document _id, _rev and _deleted=true. The data size per deleted doc depends
+on the number of revisions that CouchDB has to track plus the data size for any
+data stored in the deleted revision (this is usually relatively small, kilobytes
+perhaps, but varies based on use case). It is possible to keep audit trail data
+with a deleted document (ie. application-specific things like "deleted_by" and
+"deleted_at"). While generally this is not an issue, if the DB is still larger
+than expected, even after considering the minimum size of a deleted document,
+check to insure that the deleted document doesn't contain data not unintended
+for keeping past the deletion action. Specifically, if your client library is
+not careful, it could be storing a full copy of each document in the deleted
+revisions. For more information:
+https://issues.apache.org/jira/browse/COUCHDB-1141
```

Comment:
Should include an explanation of document conflicts as well, as those can be more of a reason why DBs after compaction are still big.

It should also explain that Fauxton will highlight when more deleted documents are in a DB than actual documents as a notice that your database may be underperforming due to an excess of deleted documents - and then point to the next question in the FAQ.